golos.publication/golos.publication.cpp

```diff
@@ -225,25 +228,28 @@ void publication::create_message(name account, std::string permlink,
     seconds_diff /= eosio::seconds(1).count();
     uint64_t delay_sec = cashout_window_param.window > seconds_diff ? cashout_window_param.window - seconds_diff : 0;
     if (!closed && delay_sec)
-        close_message_timer(account, message_id, delay_sec);
+        close_message_timer(message_id.author, id_message, delay_sec);
     else //parent is already closed or is about to
         message_table.modify(mssg_itr, _self, [&]( auto &item) { item.closed = true; });
 }
 
-void publication::update_message(name account, std::string permlink,
+void publication::update_message(structures::mssgid message_id,
                               std::string headermssg, std::string bodymssg,
                               std::string languagemssg, std::vector<structures::tag> tags,
                               std::string jsonmetadata) {
-    require_auth(account);
-    tables::message_table message_table(_self, account.value);
+    require_auth(message_id.author);
+    tables::message_table message_table(_self, message_id.author.value);
     auto message_index = message_table.get_index<"bypermlink"_n>();
-    auto message_itr = message_index.find(permlink);
-    eosio_assert(message_itr != message_index.end(), "Message doesn't exist.");
-    tables::content_table content_table(_self, account.value);
+    auto message_itr = message_index.find({message_id.permlink, message_id.ref_block_num});
+
+    if (is_close_message_by_ref_block_num(message_id.ref_block_num) || message_itr->closed)
```

Comment:
For this task, we only change message identification. Checks for actions with posts do not change yet.